subsys/bluetooth/mesh/pb_adv.c

```diff
@@ -637,7 +637,14 @@ static void gen_prov_start(struct prov_rx *rx, struct net_buf_simple *buf)
 
 	link.rx.last_seg = START_LAST_SEG(rx->gpc);
 
+	/* This (BIT(0) is set) can happen if we received a Transaction Continuation PDU, before
+	 * receiving a Transaction Start PDU (see `gen_prov_cont`). Now we received the Transaction
+	 * Start PDU and we can extract the last segment number. Knowing this, we check if
+	 * previously received segment exceeds the last segment number. If so, we reject the
+	 * Transaction Start PDU.
+	 */
```

Comment:
Not exactly related to this, but I think this seems to be incorrect?

`{ gen_prov_start, true, 3 },`, shouldn't this be `{ gen_prov_start, true, 5 },`?

Reply:
Not 5, but 4, because `min_len` exludes first octet. But, the Data field is checked inside `gen_prov_start` probably for logging, so this seems ok:
https://github.com/zephyrproject-rtos/zephyr/blob/55d3deb030bf10198b675fe6ce2effe738f9c01b/subsys/bluetooth/mesh/pb_adv.c#L611-L615